### PR TITLE
added code to cleanup references to opened jit before closing a browser

### PR DESCRIPTION
**Pull request by soichih**
This update makes sure that the online counter won't be stuck when a user closes the list page. 
Also added a few scripts to download pdf and convert them to jpg.

### CODE REVIEW

**Comment by yarikoptic** [comment]
> This update makes sure that the online counter won't be stuck when a user closes the list page.

Testing locally

I have opened one in my main chromium instance, and another in incognito -- counter remained 1.  I closed one (in nonincognito) it reset to 0 (disappeared) although I was still in jitsi in the other window.
Is counter IP specific (thus then behavior is "expected") or it just doesn't work as expected?

**Comment by soichih** [comment]
Yes, counters are IP specific. We could switch to issue a random ID that's session specific.. but I thought in most cases, using IP should work fine.. and also makes it a bit more tampering proof?? We could try switching to session ID though.

File: tools/create_thumb.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -x
```
Comment (yarikoptic) [inline tools/create_thumb.sh]: if `set -x` I would recommend to ensure that PS4 is set.  Quite often in my case, not setting it explicitly leads to side effects.  My favorite is smth like
```
$> bash -c "( PS4='[$?][$(date +%DT%T)]> '; set -x; echo 1; echo 2; )"   
[0][06/26/20T11:59:50]> echo 1
1
[0][06/26/20T11:59:50]> echo 2
2
```
so I could see when was it ran and either previous command failed.

note: ideally PRs shouldn't mix in unrelated changes -- this script could have gone to a new branch/separate PR and could have been merged without waiting for JS changes.

File: tools/download_posters.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env node
```
Comment (yarikoptic) [inline tools/download_posters.js]: note to myself -- this JS is not needed for frontend, it is to download all the posters, so if anything could come with `tools/create_thumb.sh`

File: tools/create_thumb.sh (hunk 1 above)
Comment (soichih) [reply tools/create_thumb.sh]: Oh~ that's a cool trick. I didn't know there was `set -x` specific PS configuration! You can add $LINENO!

```
The PS4 defined below in the ps4.sh has the following two codes:

$0 – indicates the name of script
$LINENO – displays the current line number within the script
```

What sort of side-effects have you ran into?

And yes, I wouldn't have mixed unrelated changes if I was doing this a bit more professionally.. :) I will move them out to the new -thumbnail you just setup.

File: tools/download_posters.js (hunk 2 above)
Comment (soichih) [reply tools/download_posters.js]: Yes, this could just be a simple shell script... with jq and for loop.

**Comment by yarikoptic** [comment]
I have tested with current deployed one -- it is the same incorrect behavior, so this shouldn't make it worse, merging

File: tools/create_thumb.sh (hunk 1 above)
Comment (yarikoptic) [reply tools/create_thumb.sh]: > What sort of side-effects have you ran into?

I think whenever my zsh had it set to some non-bash digestable value but then I was sourcing some bash script, it lead to crashes from eval'ing of that PS4.  So I just usually now make it explicit

**Comment by yarikoptic** [comment]
> Yes, counters are IP specific. We could switch to issue a random ID that's session specific...

ah cool -- that explains it.  I guess there is no need to overengineer, although it makes it a bit hard to test for correct behavior.  If you kept e.g. a list of IPs, and then removed the one (first one) "leaving" -- then it could work even from the same IP.  

Ah,  and I would not be surprised when people from some universities, in other countries in particular where IPv4 is scarce, would be coming from the same NAT gateway IP

**Comment by soichih** [comment]
> Ah, and I would not be surprised when people from some universities, in other countries in particular where IPv4 is scarce, would be coming from the same NAT gateway IP

Yes.. I thought I could get away with it.. because everyone is connected from their home office. I will see if I can switch to use the session ID.
